Update User Guide - Geoprocessing Tools section (#9370)
geosolutions-it/MapStore2: docs/user-guide/geoprocessing-tool.md

# GeoProcessing Tool

> *******************
> 

Reviewer: ```suggestion
```

> This tool allows to perform geographic operations using one or more layers containing geometry and save the result of the operation as a new layer.

Reviewer: ```suggestion
This tool allows to perform geographic operations using one or more layers and saving the result of the operations as new layers.
```

> !!! note
>     The **GeoProcessing Tool** is not included by default in the MapStore configuration but it can be configured within an [application contexts](application-context.md#configure-plugins) if needed.
>     The plugin works only if the **GeoProcessing WPS service** (available as a community extension) is properly installed in [GeoServer](http://geoserver.org/).

Reviewer: ```suggestion
!!! note
    The **GeoProcessing Tool** is not included by default in the MapStore viewer. It can be configured by editing the default viewer configuration or, from the UI, within an [application contexts](application-context.md#configure-plugins).
    The plugin works only when layers are hosted on a [GeoServer](http://geoserver.org/) where **GeoProcessing WPS service** (available as a community extension) is properly installed in 
```

>     The plugin works only if the **GeoProcessing WPS service** (available as a community extension) is properly installed in [GeoServer](http://geoserver.org/).

Reviewer: I looked for the process on dev, stable and maintainance but I couldn't find it (geoserver.org). Is the name correct? it is published only from a certain version?

> By clicking the **GeoProcessing Tool** <img src="../img/button/geoprocessing-button.jpg" class="ms-docbutton"/> button, available in the [Side Toolbar](mapstore-toolbars.md#side-toolbar), a panel opens so that the user can choose the geographic operations between **Buffer** and **Intersection**.

Reviewer: ```suggestion
By clicking the **GeoProcessing Tool** <img src="../img/button/geoprocessing-button.jpg" class="ms-docbutton"/> button, available in the [Side Toolbar](mapstore-toolbars.md#side-toolbar), a panel opens so that the user can choose the geographic operations between **Buffer** and **Intersection**.
```

<img src="../img/geoprocessing-tool/geoprocessing-panel.jpg" class="ms-docimage" width="400px"/>

## Buffer tool

> To get a layer buffer, the editor can select **Buffer** as a process, then can select a layer from the **Source Layer** option drop down menu and one of the layer's features from the **Source Feature** option drop down menu or, enable the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button that gives the possibility to select the *Feature* directly from the map.

Reviewer: ```suggestion
The buffer tool allows you to create a buffer around the input geometries. When the process **Buffer** is selected, the user will be able to :
* select a layer from the **Source Layer** option drop down menu. The dropdown will show the layers available for the process.
* select one of the layer's features from the **Source Feature**. The feaure can be selected directly by the dropdown menu or from the map, activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button.
```

<img src="../img/geoprocessing-tool/select-buffer-feature.mp4" class="ms-docimage"/>

> After entering the desired buffer **Distance**, the editor can click <img src="../img/button/run_button.jpg" class="ms-docbutton"/> to get a new buffer layer.

Reviewer: ```suggestion
* Insert the **Distance** for the buffer.

 The editor can now click <img src="../img/button/run_button.jpg" class="ms-docbutton"/> to start the process and generate the new buffer layer.
```

<img src="../img/geoprocessing-tool/run_buffer-layer.mp4" class="ms-docimage"/>

> The buffer layer is now present in the [TOC](toc.md) and visible in the map viewer.

Reviewer: ```suggestion
The buffer layer will be added to the [TOC](toc.md) and visible in the map viewer.
```

### Advanced Settings

Enabling the *Advanced options* the editor can include the following to the *Buffer* operation:

<img src="../img/geoprocessing-tool/buffer-advanced-options.jpg" class="ms-docimage" width="400px" />

> * The **Quadrant Segments**

Reviewer: ```suggestion
* The **Quadrant Segments**, that is the number of line segments used to approximate a quarter circle
```

* The **Style for the buffer end caps** choosing between `Round`, `Flat` or `Square`

## Intersection tool

> Once the **Intersection** has been chosen as a process, the editor can select a layer to intersect from the **Source Layer** option drop down menu and one of the layer's features from the **Source Feature** option drop down menu or, enable the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button that gives the possibility to select the *Feature* directly from the map.

Reviewer: ```suggestion
Once the **Intersection** has been chosen as a process, the editor can select a layer to intersect 2 layers and 2 features.
* the **Source Layer** from the drop down menu 
* one of the layer's features from the **Source Feature**. The feature can be selected directly from the drop down menu or by clicking on the map, activating the <img src="../img/button/add_marker_button.jpg" class="ms-docbutton"/> button.
```

<img src="../img/geoprocessing-tool/select-feature.mp4" class="ms-docimage"/>

In the same way it is possible to select the **Intersection Layer** and the corresponding **Intersection Feature** to obtain the new intersected layer by clicking on <img src="../img/button/run_button.jpg" class="ms-docbutton"/> button.

<img src="../img/geoprocessing-tool/run_intersection-layer." class="ms-docimage"/>

> The intersected layer is now present in the [TOC](toc.md) and visible in the map viewer.

Reviewer: ```suggestion
The new layer, result of the intersection of the features selected, will be added to the [TOC](toc.md) and visible in the map viewer.
```

<img src="../img/geoprocessing-tool/intersection-layer.jpg" class="ms-docimage" />

### Advanced Settings

Enabling the *Advanced options* the editor can include the following to the *Intersection* operation:

<img src="../img/geoprocessing-tool/intersection-advanced-options.jpg" class="ms-docimage" width="400px"/>

* The **First feature collection attribute to include**

* The **Second feature collection attribute to include**

* The **Specifies geometry computed for intersecting features** choosing between `INTERSECTION`, `FIRST` or `SECOND`

* The **Indicates whether to output feature area percentages** choosing between `False` or `True`

> * The **Indicates whether to output feature areas** choosing between `False` or `True`.

Reviewer: Could you explain the puprose of these options? options are available in the UI, so writing them here is not useful. But the meaning these options is not obvious.
